### PR TITLE
Support Ubuntu 18.04.

### PR DESCRIPTION
**Pull request by manics**
Add ubuntu 18.04 support

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -6,17 +6,27 @@
   yum:
     name: epel-release
     state: present
+  when: ansible_os_family == 'RedHat'
 
-- name: system packages | basic system utils
+# use separate yum/apt tasks because package only installs one item at a time
```
Comment (sbesson) [inline tasks/main.yml:11]: Is the main advantage of using `yum/apt` with a list rather than `package` with `with_items` installation speed?

Comment (manics) [reply tasks/main.yml:11]: Exactly, I guess it's not a big deal either way though.

**Review by sbesson — APPROVED** [review]
Barring the nitpicky discussion about `yum/apt` vs `package`, the proposed look good and the tests cover the expected behavior of the role. Unless @jburel and @joshmoore want to take a look, I am happy to see merged and tagged so that downstream roles can start consuming it. `1.0.2` or `1.1.0` ?

**Comment by joshmoore** [comment]
Also fine for it to go in. Before we roll out to too many repos, though, I'd like to hear opinions on using a "standard" include-file across the board, e.g. https://github.com/openmicroscopy/infrastructure/blob/master/ansible/ci-playbooks/roles/pkg-jdk8/tasks/main.yml

```
---
- include_tasks: debian.yml
  when: ansible_os_family == 'Debian'

- include_tasks: redhat.yml
  when: ansible_os_family == 'RedHat'

- include_tasks: freebsd.yml
  when: ansible_os_family == 'FreeBSD'

- include_tasks: windows.yml
  when: ansible_os_family == 'Windows'
```

**Comment by manics** [comment]
I think it depends on the role. If it's complicated with multiple conditional tasks then splitting them out into platform dependent files make sense. In cases where it's only a very small number of tasks splitting it into several files increases the maintenance burden since you can't see everything in one place.

**Comment by joshmoore** [comment]
Yeah, with this file's shortness, I can see what you mean, but consistency will tend to help all of us in the long run.

**Comment by manics** [comment]
If we want complete consistency we'll end up having a lot of duplication. There are roles where platform specific tasks are interleaved with common tasks, so splitting them up means either duplicating the common tasks, or having multiple platform files e.g. `1-redhat.yml`/`1-debian.yml`, `2-common.yml`, `3-redhat.yml`/`3-debian.yml`, `4-common.yml`, etc.

**Comment by joshmoore** [comment]
But package installation will fairly reliably be the first block, no?

**Comment by manics** [comment]
Yes, but if there are interleaved tasks later I don't see the advantage of only having package installation in platform specific files.

**Comment by sbesson** [comment]
Is it worth working through a concrete example of role where https://github.com/openmicroscopy/ansible-role-basedeps/pull/6#issuecomment-444047368 would apply? From my side, I can certainly conceive a top-level `tasks/main.yml` layout that would read as follows (including in cases as simple as this repository):

```
---
- include_tasks: install_{{ ansible_os_family | lower }}.yml
- include_tasks: {common_tasks}.yml
- include_tasks: {os_specific_tasks}_{{ ansible_os_family | lower }}.yml
- include_tasks: {more_common_tasks}.yml
```

NB: the example above uses `include_tasks` but there might be consideration for using `import_tasks` in some cases.

**Comment by manics** [comment]
It's most likely to occur where the application configuration differs slightly between platforms, e.g. different layout of directories, or different conventions. I can't think of an immediate example since we currently only support one platform.

**Comment by sbesson** [comment]
Merging and I will tag as `1.1.0` alongside the migration to the Galaxy ome namespace. I suspect we will need to come back to the OS layout if we start working on e.g. Ubuntu 18.04 support through the roles stack.

**Pull request merged**